tools/istio-iptables/pkg/cmd/root.go

```diff
@@ -69,25 +70,41 @@ var rootCmd = &cobra.Command{
 			}
 		}
 
-		iptConfigurator := capture.NewIptablesConfigurator(cfg, ext)
 		if !cfg.SkipRuleApply {
+			iptConfigurator := capture.NewIptablesConfigurator(cfg, ext)
 			iptConfigurator.Run()
 			if err := capture.ConfigureRoutes(cfg, ext); err != nil {
 				log.Errorf("failed to configure routes: ")
 				handleErrorWithCode(err, 1)
 			}
 		}
 		if cfg.RunValidation {
-			hostIP, err := getLocalIP()
+			hostIPs, err := getLocalIP()
 			if err != nil {
 				// Assume it is not handled by istio-cni and won't reuse the ValidationErrorCode
 				panic(err)
 			}
-			validator := validation.NewValidator(cfg, hostIP)
-
-			if err := validator.Run(); err != nil {
-				handleErrorWithCode(err, constants.ValidationErrorCode)
+			var checkIPv4, checkIPv6 bool
+			for _, hostIP := range hostIPs {
+				// it's unnecessary to check all IP, only one of IPv4 and IPv6 should be validated
```

Comment:
I don't understand this logic. Seems `checkIPv4 && checkIPv6` can never be `true` since `IPv4 || checkIPv6` will always skip things. Why not just keep the old logic, which was simpler and seems to do the same thing?

Reply:
This logic can make sure that IPv4 and IPv6 addresses can be validated in dual-stack cluster, and just validate one IPv4 and one IPv6, no more than that.  `checkIPv4 && checkIPv6` can be true in dual-stack cluster, and `checkIPv4 || checkIPv6` can skip unnecessary IP validation.